sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/flow/FlowRuleManager.java

```diff
@@ -89,6 +87,27 @@ public static void register2Property(SentinelProperty<List<FlowRule>> property)
         return rules;
     }
 
+    /**
+     * add a single FlowRule
+     *
+     * @param rule
+     * @return void
+     */
+    public static void addRule(FlowRule rule){
+        if (isValidRule(rule)) {
+            String resource = rule.getResource();
+            List<FlowRule> oldSet = flowRules.get(resource);
+            if (null != oldSet) {
+                oldSet.add(rule);
+                flowRules.put(resource, oldSet);
+                return;
+            }
+            List<FlowRule> newSet = new ArrayList<>();
+            newSet.add(rule);
+            flowRules.put(resource, newSet);
```

Comment:
All changes regarding rules SHOULD be made via the related `SentinelProperty` (and the rule change listener), as there may be additional logic. It's not allowed to modify the rule map directly.